test/parallel/test-assert.js

```diff
@@ -409,6 +409,10 @@ assert.throws(() => {
 }, /Got unwanted exception: user message/,
               'a.doesNotThrow ignores user message');
 
+assert.throws(() => {
+  assert.doesNotThrow(makeBlock(thrower, Error), 'user message');
+}, /Got unwanted exception: user message\ntest/);
```

Comment:
@Trott thank you for recommendations. I've added one more test to check if no `user message` provided. 

As for second comment about throw primitive... it really displays `undefined` as error text. It goes to another branch and do not entering in the `if` that I touched in PR. It happens probably because `util.isError` is returning `false` since string isn't good Error object. 
I'm still thinking about good way to fix it...